internal/statemachine/helper.go

```diff
@@ -1017,34 +1037,35 @@ func (stateMachine *StateMachine) updateGrub(rootfsVolName string, rootfsPartNum
 		return err
 	}
 
-	var umounts []*exec.Cmd
+	// detach the loopback device
+	teardownCmds = append(teardownCmds,
+		//nolint:gosec,G204
+		execCommand("losetup", "--detach", loopUsed),
```

Comment:
This is fine, +1, just made me think of something. Please note that I am not an experienced Go developer, so I'm fine if you say it's not the way to go. I was thinking: since the `associateLoopDevice` command basically attaches the loop device, one thing that comes to my mind would for the actual function to return the command that needs to be executed to 'clean up' when we're done. Not sure if that's a scheme that's frequent in Golang though. What do you think? This or having the detach bits as a separate helper function, so that people know that once you associate, you have to unassociate as well at some point.
Also, not very important, just overthinking things!

Reply:
I agree with you, I will do this. Moreover it will follow the same pattern used for the mount/unmount. It also makes sense to apply this to the dpkg-divert command I think.
I am not a golang expert either, but I think it looks fine to return a function to be called to cleanup things (using closure is rather idiomatic I think).